TestON/drivers/common/cli/onosclidriver.py

```diff
@@ -1256,16 +1256,26 @@ def addMultipointToSinglepointIntent(
             main.cleanup()
             main.exit()
 
-    def removeIntent( self, intentId ):
-        """
-        Remove intent for specified intent id
-
+    def removeIntent( self, intentId, app = 'org.onosproject.cli',
+        purge = False, sync = False ):
+        """
+        Remove intent for specified application id and intent id
+        Optional args:- 
+        -s or --sync: Waits for the removal before returning
+        -p or --purge: Purge the intent from the store after removal  
+        
         Returns:
             main.False on error and
             cli output otherwise
         """
         try:
-            cmdStr = "remove-intent org.onosproject.cli " + str( intentId )
+            cmdStr = "remove-intent "
+            if purge:
+                cmdStr += " -p"
+            if sync:
+                cmdStr += " -s"
+
+            cmdStr += " " + app + " " + str( intentId )
```

Comment:
technically the type of app could be anything if it is passed, so it would be safer to typecast it to int. But if they are passing something besides a string, they are probably passing the wrong thing anyways.